Flaky build

The functional tests of the SignalR client occasionally [fail](http://aspnetci/viewLog.html?tab=buildLog&buildTypeId=XPlat_Windows_Win2008r2_Universe&buildId=443617&_focus=110826) with the following error:

```
[06:40:38]	SignalR | dotnet | It was not possible to find any compatible framework version
[06:40:38]	SignalR | The specified framework 'Microsoft.NETCore.App', version '2.1.0-preview3-26331-01' was not found.
[06:40:38]	SignalR | - Check application dependencies and target a framework version installed at:
[06:40:38]	SignalR | C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\
[06:40:38]	SignalR | - Installing .NET Core prerequisites might help resolve this problem:
[06:40:38]	SignalR | http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
[06:40:38]	SignalR | - The .NET Core framework and SDK can be installed from:
[06:40:38]	SignalR | https://aka.ms/dotnet-download
[06:40:38]	SignalR | - The following versions are installed:
[06:40:38]	SignalR | 2.0.0 at [C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\shared\Microsoft.NETCore.App]
[06:40:38]	SignalR | 2.0.5 at [C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\shared\Microsoft.NETCore.App]
[06:40:38]	SignalR | 2.1.0-preview2-26313-01 at [C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\shared\Microsoft.NETCore.App]
[06:40:38]	SignalR | 2.1.0-preview2-26403-06 at [C:\BuildAgent\work\33bdfc1cae7b2a38\.dotnet\x64\shared\Microsoft.NETCore.App]
[06:40:38]	SignalR |
[06:40:38]	SignalR | dotnet process exited with code: 2147516566
[06:40:38]	SignalR | Terminating dotnet process
[06:40:38]	SignalR | npm ERR! Test failed. See above for more details.
[06:40:38]	SignalR | npm ERR! code ELIFECYCLE
[06:40:38]	SignalR | npm ERR! errno 1
[06:40:38]	SignalR | npm ERR! functionaltests@1.0.0 ci-test: `ts-node --project ./selenium/tsconfig.json ./selenium/run-ci-tests.ts "--configuration" "Release"`
[06:40:38]	SignalR | npm ERR! Exit status 1
[06:40:38]	SignalR | npm ERR!
[06:40:38]	SignalR | npm ERR! Failed at the functionaltests@1.0.0 ci-test script.
[06:40:38]	SignalR | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[06:40:39]	SignalR |
[06:40:39]	SignalR | npm ERR! A complete log of this run can be found in:
[06:40:39]	SignalR | npm ERR! C:\Users\asplab\AppData\Roaming\npm-cache\_logs\2018-04-09T13_40_39_856Z-debug.log
```

Future instances of this failure can be found [here](http://aspnetci/httpAuth/app/rest/builds?locator=tag:NPMFramework,sinceDate:201710010T000000-0001&fields=build(startDate,buildTypeId,status,branchName,webUrl,tags)).

--- a/build/repo.targets
+++ b/build/repo.targets
@@ -33,6 +33,7 @@
 
   <Target Name="RunBrowserTests">
     <Message Text="Running TypeScript client Browser tests" Importance="high" />
+    <Exec Command="npm run build" WorkingDirectory="$(RepositoryRoot)clients/ts/FunctionalTests" IgnoreStandardErrorWarningFormat="true" />
     <Exec Command="npm run ci-test -- --configuration $(Configuration)" WorkingDirectory="$(RepositoryRoot)clients/ts/FunctionalTests" IgnoreStandardErrorWarningFormat="true" />
   </Target>
 

--- a/clients/ts/FunctionalTests/package.json
+++ b/clients/ts/FunctionalTests/package.json
@@ -21,7 +21,8 @@
     "build:tsc": "node ../node_modules/typescript/bin/tsc --project ./tsconfig.json",
     "build:rollup": "node ../node_modules/rollup/bin/rollup -c",
     "pretest": "npm run build",
-    "test": "dotnet build && ts-node --project ./selenium/tsconfig.json ./selenium/run-tests.ts",
+    "test": "dotnet build && npm run test-only",
+    "test-only": "ts-node --project ./selenium/tsconfig.json ./selenium/run-tests.ts",
     "ci-test": "ts-node --project ./selenium/tsconfig.json ./selenium/run-ci-tests.ts"
   },
   "author": "",

--- a/clients/ts/FunctionalTests/selenium/run-ci-tests.ts
+++ b/clients/ts/FunctionalTests/selenium/run-ci-tests.ts
@@ -80,9 +80,9 @@ if (configuration) {
     args.push("--configuration");
     args.push(configuration);
 }
-if (verbose) {
+
     args.push("--verbose");
-}
+
 
 let command = "npm";
 

--- a/clients/ts/FunctionalTests/selenium/run-ci-tests.ts
+++ b/clients/ts/FunctionalTests/selenium/run-ci-tests.ts
@@ -74,8 +74,8 @@ if (!existsSync(chromeBinary)) {
     console.log(`Using Google Chrome Browser from '${chromeBinary}`);
 }
 
-// Launch the tests
-const args = ["test", "--", "--raw", "--chrome", chromeBinary];
+// Launch the tests (we know the CI already built, so run the 'test-only' script)
+const args = ["run", "test-only", "--", "--raw", "--chrome", chromeBinary];
 if (configuration) {
     args.push("--configuration");
     args.push(configuration);

